### PR TITLE
Include TrustLevel in InfoNode query

### PR DESCRIPTION
**Pull request by GiasemiSh**
closes https://github.com/QMSTR/qmstr-all/issues/97

This PR just creates the query which uses the trust level and returns the most trustworthy data.
We do not query for the data type yet.

Query is not yet used by reporters. We discussed a restructure of the code, to extract bom as a library, and make calls to the db on specific node types (not just query for a package node and traverse on it).

### CODE REVIEW

**Comment by hemarkus** [comment]
closes #276

**Comment by hemarkus** [comment]
@GiasemiSh this needs to be rebased

File: lib/go-qmstr/database/infoNode.go [hunk 1]
```diff
@@ -72,6 +72,76 @@ func (db *DataBase) AddInfoNodes(nodeID string, infonodes ...*service.InfoNode)
...
+		return nil, nil
+	}
+
+	realData := []string{}
+
+	for _, info := range infoData {
+		// infoData contains all the infodata attached to the filenode (with the declared info type)
+		// but the query returns only the most trusted analyzer connected to the infonodes
+		// So only info nodes with an analyzer attached are the trustworthy data
+		if len(info.Analyzer) != 0 {
```
Comment (hemarkus) [inline lib/go-qmstr/database/infoNode.go]: So it should be > not !=

**Comment by GiasemiSh** [comment]
@hemarkus Works for me. Can you please rebuild the snapshots and retry?

File: lib/go-qmstr/database/infoNode.go (hunk 1 above)
Comment (GiasemiSh) [reply lib/go-qmstr/database/infoNode.go]: done

**Comment by GiasemiSh** [comment]
@hemarkus Now it should produce author and license information

**Pull request merged**